### PR TITLE
mommy has lots to explain~

### PR DESCRIPTION
**Pull request by Gankra**

### CODE REVIEW

File: book/src/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  --> src\main.rs:20:6
+   |
+20 | mods passes;
+   |      ^^^^^^ expected one of `!` or `::`
+
+error: could not compile `bappy-script` (bin "bappy-script") due to previous error
+mommy knows her little girl can do better~ 💞
+```
+
+By default mommy is kind to her little girl, but she can become anything you wants~
```
Comment (workingjubilee) [inline book/src/README.md:23]:
```suggestion
By default mommy is kind to her little girl, but she can become anything you want~
```

File: book/src/contributing/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Contributing
+
+These are the technical rules of mommy club~
+
+* mommy as a project is always going to *default* to being mommy, and *default* to loving her little girl, even if she also does everything she can to be customized for people who don't want that~ 💖
```
Comment (workingjubilee) [inline book/src/contributing/README.md:5]:
```suggestion
* mommy as a project is always going to *default* to being mommy, and *default* to loving her little girl, even if she also does everything she can to be customizable for people who don't want that~ 💖
```

File: book/src/contributing/concepts.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+# Concepts
+
+At its heart, mommy exists to invoke cargo, check if the result was a success or failure, and print an appropriate message.
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:3]:
```suggestion
At her heart, mommy exists to invoke cargo, check if the result was a success or failure, and print an appropriate message~
```

File: book/src/contributing/concepts.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+Printing that message involves reading a bunch of configuration from env-vars to define pools of values to randomly select from, then:
+
+1. randomly [deciding what mood the response should have](#moods)
+2. using the success/failure state to select which pool of answer from that mood to use (positive, negative, overflow)
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:7]:
```suggestion
2. use the success/failure state to select which pool of answers from that mood to use (positive, negative, overflow)
```

File: book/src/contributing/concepts.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+3. randomly selecting a message from that pool
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:8]:
```suggestion
3. randomly select a message from that pool
```

File: book/src/contributing/concepts.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+# Concepts
+
+At its heart, mommy exists to invoke cargo, check if the result was a success or failure, and print an appropriate message.
+Printing that message involves reading a bunch of configuration from env-vars to define pools of values to randomly select from, then:
+
+1. randomly [deciding what mood the response should have](#moods)
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:6]:
```suggestion
1. randomly [decide what mood the response should have](#moods)
```

File: book/src/contributing/concepts.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+2. using the success/failure state to select which pool of answer from that mood to use (positive, negative, overflow)
+3. randomly selecting a message from that pool
+
+cargo-mommy has 3 major components~
+
+1. [responses.json, where data lives](https://github.com/Gankra/cargo-mommy/blob/main/responses.json)
+2. [build.rs, that digests that data and coverts it to rust code](https://github.com/Gankra/cargo-mommy/blob/main/build.rs)
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:13]:
```suggestion
2. [build.rs, that digests that data and converts it to rust code](https://github.com/Gankra/cargo-mommy/blob/main/build.rs)
```

File: book/src/contributing/concepts.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    }
+}
+```
+
+
+
+
+### Moods
+
+Moods contain [pools of messages](#message-pools) with a particular feeling/intensity. This was originally introduced to allow the user to opt into [nsfw functionality](../customize/nsfw.md), but exists more generically than that, and we'd be happy to have more sfw moods like "furry" or whatever.
```
Comment (workingjubilee) [inline book/src/contributing/concepts.md:49]:
```suggestion
Moods contain [pools of messages](#message-pools) with a particular feeling/intensity. This was originally introduced to allow the user to opt into [nsfw functionality](../customize/nsfw.md). Its functionality is more general than that, and we'd be happy to have more sfw moods like "furry" or whatever.
```

File: book/src/customize/true-roles.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+* folks who *really* don't want to say/see "mommy" ever again
+* folks who want to maintain several independent sets of configuration (perhaps for plural reasons)
+
+Specifically if you change the True Role from "mommy" to e.g. "daddy", the following changes will occur:
+
+* You will be able to invoke it as `cargo daddy`
+* Instead of reading env vars like `CARGO_MOMMYS_MOODS` for config, it will read `CARGO_DADDYS_MOODS` (note the extra "S"!)
+* If `CARGO_{TRUE_ROLE}S_ROLE` isn't set, it will default to the True Role
+
+The value "daddy" is abitrary here, you can pick any value. Make yourself a `cargo burger-chimes` if you want!
```
Comment (workingjubilee) [inline book/src/customize/true-roles.md:25]:
```suggestion
The value "daddy" is arbitrary here, you can pick any value. Make yourself a `cargo burger-chimes` if you want!
```

File: .gitignore [hunk 10]
```diff
@@ -1,4 +1,5 @@
 /target
-
 # Generated by `oranda generate ci`
-public/
\ No newline at end of file
+public/
+# Generated by mdbook
+book/book/
```
Comment (workingjubilee) [inline .gitignore:5]: And as long as I'm here
```suggestion
book/book/

```

**Review by workingjubilee — COMMENTED** [review]
gonna need a `mommy's-little-copyeditor@jubilee.dev` huh

**Comment by workingjubilee** [comment]
Note: Either in the main README.md, or a README for this particular folder, contributors should be advised on what an "mdbook" is (and prompted to `cargo install mdbook` and consult `mdbook --help`) in the likely event they have never seen one. This will allow them to know how to build the documentation. It seems reasonable to assume they have access to the default Rust tooling but, although you would not know this by the frequency of the community use, mdbook is technically not such: it is a user's binary crate.

**Comment by Gankra** [comment]
So strictly speaking they actually want to install `oranda` and run `oranda dev` (which uses mdbook as a library and patches several bugs so that it can unify the themes).

Pushed up many more docs.

File: book/src/contributing/adding-moods.md [hunk 11]
```diff
@@ -1,3 +1,38 @@
 # Adding A New Mood
 
-TODO: add proper guide, see [concepts for now](./concepts.md)
+You shouldn't need any code to add a mood, just update [repsonses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
```
Comment (workingjubilee) [inline book/src/contributing/adding-moods.md:3]:
```suggestion
You shouldn't need any code to add a mood, just update [responses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
```

File: book/src/contributing/adding-moods.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
+# Adding A New Mood
+
+You shouldn't need any code to add a mood, just update [repsonses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
+
+## Editing responses.json
+
+* Add [the new mood to reponses.json's "moods" object](./concepts.md#responsesjson)~
```
Comment (workingjubilee) [inline book/src/contributing/adding-moods.md:7]:
```suggestion
* Add [the new mood to responses.json's "moods" object](./concepts.md#responsesjson)~
```

File: book/src/contributing/adding-variables.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
+# Adding A New Variable
+
+You shouldn't need any code to add a new variable, just update [repsonses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
+
+
+## Editing responses.json
+
+* Add [the new variable to reponses.json's "vars" object](./concepts.md#responsesjson)~
```
Comment (workingjubilee) [inline book/src/contributing/adding-variables.md:8]:
```suggestion
* Add [the new variable to responses.json's "vars" object](./concepts.md#responsesjson)~
```

File: book/src/contributing/adding-variables.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
+# Adding A New Variable
+
+You shouldn't need any code to add a new variable, just update [repsonses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
```
Comment (workingjubilee) [inline book/src/contributing/adding-variables.md:3]:
```suggestion
You shouldn't need any code to add a new variable, just update [responses.json](https://github.com/Gankra/cargo-mommy/blob/main/responses.json) and the docs~
```

File: book/src/customize/moods.md [hunk 15]
```diff
@@ -1,3 +1,66 @@
 # Moods
 
-TODO: flesh this out, see [roles and pronouns for now](./roles-and-pronouns.md)
+Moods are a system for changing the flavour/vibe of mommy's messages~ 💕
```
Comment (workingjubilee) [inline book/src/customize/moods.md:3]: the vibe, huh

**Comment by workingjubilee** [comment]
> So strictly speaking they actually want to install oranda and run oranda dev (which uses mdbook as a library and patches several bugs so that it can unify the themes).

Indeed. It just seems important to make sure to document that mdbook is the... institution? in use.

File: book/src/contributing/releasing.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+## Updating The Release Process
+
+The cargo-release step was created by hand-copying some config values that Gankra likes to use in her projects. There shouldn't be any need to mess with it.
+
+The cargo-dist step was created by locally running `cargo dist init`, selecting some targets/installers in the interactive prompts, and committing the results. It can be updated by running `cargo dist init` again.
+
+The oranda step was created by running `oranda generate ci`, selecting some of the options, and committing the results. A [custom oranda.json](https://github.com/Gankra/cargo-mommy/blob/main/oranda.json) was then added to fix path-prefix with github-pages, and to pick a non-default theme. The CI is setup to always use the "latest" oranda, so it's self-updating and may never need to be touched. Any changes that are desired will probably be changes to oranda.json.
+
+The Github Pages step was doing in the Github settings UI for cargo-mommy. The website it deploys to is Gankra's.
```
Comment (workingjubilee) [inline book/src/contributing/releasing.md:50]: Not entirely sure if this is the right edit? It seems good to qualify an explicit reference to the repository per se, even if this is easily inferred, but I don't know if the tense is right.
```suggestion
The Github Pages step was done in the Github Settings UI for the cargo-mommy repo. The website it deploys to is Gankra's.
```

**Pull request merged**